consensus/core/src/commit_observer.rs

```diff
@@ -37,6 +37,8 @@ pub(crate) struct CommitObserver {
     sender: UnboundedSender<CommittedSubDag>,
     /// Persistent storage for blocks, commits and other consensus data.
     store: Arc<dyn Store>,
+    /// The last committed sub dag
+    last_sub_dag: Option<CommittedSubDag>,
```

Comment:
nit: instead of keeping the entire last committed subdag in memory what if we just kept the commit index + timestamp?